packages/dappmanager/src/modules/migrations/addAliasToRunningContainers.ts

```diff
@@ -26,140 +26,176 @@ const dncoreNetworkName = params.DNP_PRIVATE_NETWORK_NAME;
  */
 export async function addAliasToRunningContainers(): Promise<void> {
-  for (const container of await listContainers()) {
-    const containerName = container.containerName;
-    const alias = getPrivateNetworkAlias(container);
-
-    try {
-      // Info from docker inspect and compose file might be not-syncrhnonyzed
-      // So this function must be before the check hasAlias()
-      migrateCoreNetworkAndAliasInCompose(container, alias);
-
-      const currentEndpointConfig = await getDnCoreNetworkContainerConfig(
-        containerName
-      );
-      if (hasAlias(currentEndpointConfig, alias)) continue;
-      const endpointConfig: Partial<Dockerode.NetworkInfo> = {
-        ...currentEndpointConfig,
-        Aliases: [...(currentEndpointConfig?.Aliases || []), alias]
-      };
-
-      // Wifi and VPN containers needs a refresh connect due to its own network configuration
-      if (
-        container.containerName === params.vpnContainerName ||
-        container.containerName === params.wifiContainerName
-      ) {
-        await shell(`docker rm ${containerName} --force`);
-        await dockerComposeUp(
-          getPath.dockerCompose(container.dnpName, container.isCore)
-        );
-      } else {
-        await dockerNetworkDisconnect(dncoreNetworkName, containerName);
-        await dockerNetworkConnect(
-          dncoreNetworkName,
-          containerName,
-          endpointConfig
-        );
-      }
-      logs.info(`Added alias to running container ${container.containerName}`);
-    } catch (e) {
-      logs.error(`Error adding alias to container ${containerName}`, e);
-    }
+  try {
+    const containers = await listContainers();
+    await addAliasToGivenContainers(containers);
+  } catch (error) {
+    logs.error('Error adding alias to running containers:', error);
   }
 }
 
-/** Return true if endpoint config exists and has alias */
-function hasAlias(
-  endpointConfig: Dockerode.NetworkInfo | null,
-  alias: string
-): boolean {
-  return Boolean(
-    endpointConfig &&
-      endpointConfig.Aliases &&
-      Array.isArray(endpointConfig.Aliases) &&
-      endpointConfig.Aliases.includes(alias)
-  );
+export async function addAliasToGivenContainers(containers: PackageContainer[]): Promise<void> {
+  for (const container of containers) {
+
+    const isMainOrMonoService = container.isMain ?? false; // Set a default value of false if isMain is undefined
+    const service = {
+      serviceName: container.serviceName,
+      dnpName: container.dnpName,
+      isMain: isMainOrMonoService, // Add the isMain property here
+    };
+    const aliases = getPrivateNetworkAliases(service)
+    
+    // Adds aliases to the compose file that generated the container
+    migrateCoreNetworkAndAliasInCompose(container, aliases);
+
+    // Adds aliases to the container network
+    for (const alias of aliases) {
```

Comment:
You could try to avoid the loop here.

1. Get the current alias in config
2. Compare those to the new ones and leave only those to add
3. Add them at the same time of possible